source/exe/signal_action.h

```diff
@@ -47,7 +49,9 @@ namespace Envoy {
 class SignalAction : NonCopyable {
 public:
   SignalAction()
-      : guard_size_(sysconf(_SC_PAGE_SIZE)), altstack_size_(guard_size_ * 4), altstack_(nullptr) {
+      : guard_size_(sysconf(_SC_PAGE_SIZE)),
+        altstack_size_(std::max(guard_size_ * 4, static_cast<size_t>(MINSIGSTKSZ))),
```

Comment:
Ah, yes, all of this looks correct. Guard pages are always outside the altstack_size_, and MINSIGSTKSZ should always be a multiple of page size.